### ISSUE TITLE
[🔷 Feature request ]: XOR multiple inputs

### ISSUE THREAD

**Issue by matiasvlevi** [matiasvlevi/Dann#22]
### Feature
A function that would create an XOR dataset with `X` number of inputs.

### Type
- [ ] Dann
- [ ] Matrix
- [ ] Layer
- [ ] Activation functions
- [ ] Loss functions
- [ ] Pool functions
- [x] Datasets
- [ ] Documentation
- [ ] tests & examples
- [ ] Other 

### Description
We currently have a static 2 input XOR dataset for testing/examples purposes. We also have a `makeBinary` function that creates a dataset of binary digits with `X` bits, so you can create a custom dataset to test a neural network. What if XOR  had a similar function allowing for the creation of a 3 or 4 input XOR.

### Examples

```js
const dataset = makeXOR(3);
console.log(dataset);
```

```
{
 {
  input: [0, 0, 0],
  output: [0]
 },
 {
  input: [0, 0, 1],
  output: [1]
 },
 {
  input: [0, 1, 0],
  output: [1]
 },
 {
  input: [0, 1, 1],
  output: [0]
 },
//...
 {
  input: [1, 1, 1],
  output: [1]
 },
}
```

This is a 3 input XOR table for reference 

<img src="https://external-content.duckduckgo.com/iu/?u=https%3A%2F%2Farduinotricks.com%2Fwp-content%2Fuploads%2F2018%2F12%2FXOR-3-input.jpg&f=1&nofb=1" alt="drawing" width="380"/>

**Comment by shirsho1106** [comment]
I and @Labnann want to complete your project. We are complete beginners, want to make ourselves roll on.

**Comment by matiasvlevi** [comment]
Nice to hear! Everyone is welcome! Do you want to be assigned to this issue?

**Comment by Labnann** [comment]
Yup! We two would like to be assigned.

**Comment by Labnann** [comment]
Hello! Is it ok to require a js file from builds? 

We would like to use your makeBinary to keep the code simple.

**Comment by matiasvlevi** [comment]
We can't require JS files in builds because the built file is read either by a browser or node. It would work with node in theory, but wouldn't within a browser. The `src` directory is also ignored for the npm package so requiring a file from the `src` directory wouldn't exist in the npm package. In this case, even with node, this wouldn't work.

The alternative to this is to place the file in a directory that is going to be concatenated in the built files just like `src/core/datasets` when `npm run build` is run. Which essentially acts as if you had hardcoded the feature in the built file (You can check `src/concatConfig.js` to see which files are concatenated in which order)

It's also important to note that you need to reference the method in the node module exports (You can do so in `src/io/exports.js`) so the feature would be available in the npm package.

Feel free to ask if you have any more questions!

**Comment by Labnann** [comment]
Thanks for the info! Really good to learn.

Can you look into this please?

We are getting errors with `npm run test` and `npm run build` even when there is no changes we made (when our clone is equal to the origin). 

`npm run test`:
![image](https://user-images.githubusercontent.com/55809005/123524399-d7233700-d6eb-11eb-8d96-beaaaa162929.png)

`npm run build`:
![image](https://user-images.githubusercontent.com/55809005/123524435-089c0280-d6ec-11eb-8c3f-8164cf713c46.png)

I am not sure why this is happening. I am using node 14.17.1 by the way, in windows.

**Comment by matiasvlevi** [comment]
Have you installed the dev dependencies? You can do so with `npm install --only=dev`. I'm still looking in to it in case that was not the issue.

**Comment by matiasvlevi** [comment]
I saw some people use with this specific error use `npm cache clean --force` to clear cache, then remove the `node_modules` & `package-lock.json`. They then reinstall all dependencies with `npm install`.
